### PR TITLE
Better proposed APIs desc

### PR DESCRIPTION
**Pull request by matzew**
Fixes SRVCOM-2668

## Proposed Changes

- offers more meaningful description of our provided APIs

### CODE REVIEW

**Comment by ReToCode** [comment]
I like it.
/cc @rhuss

**Comment by Preeticp** [comment]
LGTM

File: olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -214,7 +214,7 @@ spec:
             displayName: Version
             path: version
         version: v1beta1
-      - description: Represents an installation of a particular version of Knative Kafka components
+      - description: Apache Kafka focused extensions for Knative Eventing
```
Comment (Cali0707) [inline olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml:217]:
```suggestion
      - description: Apache Kafka based extensions for Knative Eventing
```
I feel like "Apache Kafka _focused_ extensions" implies that the extensions are focused on targetting Kafka, while "Apache Kafka _based_ extensions" implies that the extensions are based on Kafka.

File: templates/csv.yaml [hunk 2]
```diff
@@ -216,7 +216,7 @@ spec:
             displayName: Version
             path: version
         version: v1beta1
-      - description: Represents an installation of a particular version of Knative Kafka components
+      - description: Apache Kafka focused extensions for Knative Eventing
```
Comment (Cali0707) [inline templates/csv.yaml:219]:
```suggestion
      - description: Apache Kafka based extensions for Knative Eventing
```

**Review by Cali0707 — COMMENTED** [review]
@matzew feel free to ignore if you disagree, just my 2 cents on the wording of the Kafka bits

**Comment by matzew** [comment]
I think the text for Serving is a bit long here. I think I tend to have a more crisp description, as smaller text is easier to parse/read... :man_shrugging: 

Screenshot from 2023-09-08 08-19-12

**Comment by matzew** [comment]
Screenshot from 2023-09-11 09-42-53

Latest updates ^

File: olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -214,7 +214,7 @@ spec:
             displayName: Version
             path: version
         version: v1beta1
-      - description: Represents an installation of a particular version of Knative Kafka components
+      - description: Additionl extensions based on Apache Kafka for production-ready Knative Eventing deployments
```
Comment (pierDipi) [inline olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml:217]:
```suggestion
      - description: Additional extensions based on Apache Kafka for production-ready Knative Eventing deployments
```

File: templates/csv.yaml [hunk 4]
```diff
@@ -216,7 +216,7 @@ spec:
             displayName: Version
             path: version
         version: v1beta1
-      - description: Represents an installation of a particular version of Knative Kafka components
+      - description: Additionl extensions based on Apache Kafka for production-ready Knative Eventing deployments
```
Comment (pierDipi) [inline templates/csv.yaml:219]:
```suggestion
      - description: Additional extensions based on Apache Kafka for production-ready Knative Eventing deployments
```

File: olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml (hunk 3 above)
Comment (pierDipi) [inline olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml:217]: what do you think about this instead?
```suggestion
      - description: Knative Kafka is the Knative Eventing extensions for Apache Kafka. It allows to trigger services based on Apache Kafka topics and create your event-driven applications with a simple HTTP interface with the the underlying reliability and efficiency of Apache Kafka.
```

Comment (matzew) [reply olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml]: that's quite a mouthful. How about:

> Knative Kafka is the Knative Eventing extensions for Apache Kafka. Create event-driven applications with a simple HTTP interface while leveraging the underlying reliability and efficiency of Apache Kafka

**Comment by pierDipi** [comment]
LGTM

**Comment by Cali0707** [comment]
LGTM

**Comment by Preeticp** [comment]
LGTM

**Comment by rhuss** [comment]
I would shorten the Serving description to:

> Knative Serving simplifies application deployment, enables scaling from and to zero replicas, and enables advanced traffic routing use-cases.

* There is no such thing like a "severless container"
* Be explicit what to scale

**Comment by rhuss** [comment]
Some thoughts on the descriptions to make them more concise and aligned in the format (e.g. the Knative Kafka description is not in the same style as the others:

* Knative Serving: A platform for streamlined app deployment, traffic-based auto-scaling from zero to N, and traffic-split rollouts.
* Knative Eventing: An OpenShift-native EDA runtime platform that leverages Cloud Events with a simple HTTP interface.
* Knative Kafka: An extension to Knative Eventing, merging HTTP accessibility with Apache Kafka's proven efficiency and reliability.

I believe these very condensed and concise descriptions highlight the core features. Is there anything missing?

**Comment by matzew** [comment]
Latest proposal....

Screenshot from 2023-09-12 16-34-47

**Review by rhuss — COMMENTED** [review]
/lgtm

thanks!

**Comment by nainaz** [comment]
Love It!

**Pull request merged**